Python/Product/PythonTools/PythonTools/Options/PythonDebuggingOptionsControl.cs

```diff
@@ -40,10 +42,47 @@ public partial class PythonDebuggingOptionsControl : UserControl {
-            _showVariablesClassComboBox.SelectedItem = pyService.DebuggerOptions.VariablePresentationForClasses;
-            _showVariablesFunctionComboBox.SelectedItem = pyService.DebuggerOptions.VariablePresentationForFunctions;
-            _showVariablesProtectedComboBox.SelectedItem = pyService.DebuggerOptions.VariablePresentationForProtected;
-            _showVariablesSpecialComboBox.SelectedItem = pyService.DebuggerOptions.VariablePresentationForSpecial;
+            if (pyService.DebuggerOptions.VariablePresentationForClasses == PresentationMode.Group) {
+                _showVariablesClassComboBox.SelectedIndex = 0;
+            } else if (pyService.DebuggerOptions.VariablePresentationForClasses == PresentationMode.Hide) {
+                _showVariablesClassComboBox.SelectedIndex = 1;
+            } else if (pyService.DebuggerOptions.VariablePresentationForClasses == PresentationMode.Inline) {
+                _showVariablesClassComboBox.SelectedIndex = 2;
+            } else {
+                _showVariablesClassComboBox.SelectedIndex = 0;
+            }
+
+            if (pyService.DebuggerOptions.VariablePresentationForFunctions == PresentationMode.Group) {
+                _showVariablesFunctionComboBox.SelectedIndex = 0;
+            } else if (pyService.DebuggerOptions.VariablePresentationForFunctions == PresentationMode.Hide) {
+                _showVariablesFunctionComboBox.SelectedIndex = 1;
+            } else if (pyService.DebuggerOptions.VariablePresentationForFunctions == PresentationMode.Inline) {
+                _showVariablesFunctionComboBox.SelectedIndex = 2;
+            } else {
+                _showVariablesFunctionComboBox.SelectedIndex = 0;
+            }
+
+            if (pyService.DebuggerOptions.VariablePresentationForProtected == PresentationMode.Group) {
+                _showVariablesProtectedComboBox.SelectedIndex = 0;
+            } else if (pyService.DebuggerOptions.VariablePresentationForProtected == PresentationMode.Hide) {
+                _showVariablesProtectedComboBox.SelectedIndex = 1;
+            } else if (pyService.DebuggerOptions.VariablePresentationForProtected == PresentationMode.Inline) {
+                _showVariablesProtectedComboBox.SelectedIndex = 2;
+            } else {
+                _showVariablesProtectedComboBox.SelectedIndex = 0;
+            }
+
+            if (pyService.DebuggerOptions.VariablePresentationForSpecial == PresentationMode.Group) {
+                _showVariablesSpecialComboBox.SelectedIndex = 0;
+            } else if (pyService.DebuggerOptions.VariablePresentationForSpecial == PresentationMode.Hide) {
+                _showVariablesSpecialComboBox.SelectedIndex = 1;
+            } else if (pyService.DebuggerOptions.VariablePresentationForSpecial == PresentationMode.Inline) {
+                _showVariablesSpecialComboBox.SelectedIndex = 2;
+            } else {
+                _showVariablesSpecialComboBox.SelectedIndex = 0;
+            }
+
+
```

Comment:
To avoid repetition here and below, it might be better to use WinForms data binding instead. If you create a mapping like so:
```cs
var presentationModes = new Dictionary<PresentationMode, string> {
   { PresentationMode.Group, Strings.VariablePresentation_Group },
    ...
};
```
Then you can bind every combobox in the same exact way:
```cs
foreach (var comboBox in new[] { _showVariablesClassComboBox, ...  }) {
  comboBox.DataSource = presentationModes;
  comboBox.ValueMember = "Key";
  comboBox.DisplayMember = "Value";
}
```
And use `SelectedValue` to get/set the selection for each:
```cs
_showVariablesClassComboBox.SelectedValue = pyService.DebuggerOptions.VariablePresentationForClasses;
...
```

Reply:
ok thank you! That makes the code much cleaner. I will update that. 😊

Reply:
@int19h Hi Pavel, assigning dictionary to `comboBox.DataSource` is giving me "Complex DataBinding accepts as a data source either an IList or an IListSource" error. So I had to convert the dictionary to a `BindingSource` to make it work.

Reply:
@int19h oh wait, I forgot to switch the list of KeyValuePair back to dictionary. Let me change that real quick!